File: elixir/space-age/lib/space_age.ex
defmodule SpaceAge do
  @type planet ::
          :mercury
          | :venus
          | :earth
          | :mars
          | :jupiter
          | :saturn
          | :uranus
          | :neptune
  @seconds_on_earth 31_557_600.0

  defp planet_period(:mercury), do: 0.2408467
  defp planet_period(:venus), do: 0.61519726
  defp planet_period(:earth), do: 1.0
  defp planet_period(:mars), do: 1.8808158
  defp planet_period(:jupiter), do: 11.862615
  defp planet_period(:saturn), do: 29.447498
  defp planet_period(:uranus), do: 84.016846
  defp planet_period(:neptune), do: 164.79132
  defp round_to_two_decimals(value), do: round(value * 100.0) / 100.0

  @spec age_on(planet, pos_integer) :: float
  def age_on(planet, seconds) do
    age_on_planet = seconds / planet_period(planet) / @seconds_on_earth
    round_to_two_decimals(age_on_planet)
  end
end

File: r/space-age/space-age.R
seconds_in_earth_year = 31557600

orbital_periods_in_earth_years = c(
  mercury = 0.2408467,
  venus = 0.61519726,
  earth = 1.0,
  mars = 1.8808158,
  jupiter = 11.862615,
  saturn = 29.447498,
  uranus = 84.016846,
  neptune = 164.79132
)

space_age <- function(seconds, planet) {
  round(seconds / seconds_in_earth_year / orbital_periods_in_earth_years[[planet]], digits = 2)
}
